test/Base_Tests/src/Semantic/Multi_Value_As_Type_Refinement_Spec.enso

```diff
@@ -210,21 +213,22 @@ add_specs suite_builder =
             new_ab.a_method . should_equal "A method"
             new_ab.b_method . should_equal "B method"
 
-            # But unchecked variant should keep both types and not hide anything
+            # dispatching method on self casts to Self type and always hides other types
             ab.a_id_unchecked . is_a A . should_be_true
             ab.a_id_unchecked . is_a B . should_be_true
             ab.a_id_unchecked.a_method . should_equal "A method"
-            ab.a_id_unchecked.b_method . should_equal "B method"
+            Test.expect_panic No_Such_Method (ab.a_id_unchecked.b_method)
```

Comment:
As a result of 31967674c498941a18d54524cc8699f67476b2bc we have to make sure that dispatching a method on `self` moves the type of the method to the `firstDispatch`. This is done by https://github.com/enso-org/enso/pull/12192/files#r1935731256, but:
- right now the reordering hides all the other types
- semantically it is not necessary to hide the other types
- we could keep them visible

It is just simpler to hide the other types as there is an implementation doing that available right now. Opinions? Wanna better behavior at no slowdown? CCing @radeusgd.

Reply:
I'm not 100% sure yet as all of this has many implications that only come out in practice.

But at first, I think that hiding actually seems consistent with type checking requirements - inside of method `A.foo self` I actually do want to assume that `self:A` - so it makes sense for other 'parts' to be hidden.